cob_light/common/include/signalRightMode.h

```diff
@@ -0,0 +1,75 @@
+#ifndef SIGNAL_RIGHT_H
+#define SIGNAL_RIGHT_H
+
+#include <vector>
+
+#include "mode.h"
+
+class SignalRight : public Mode
+{
+public:
+  SignalRight(color::rgba color, uint16_t nLeds, uint8_t priority = 0, double freq = 0.25, uint8_t pulses = 0,
+              double timeout = 0)
+    : Mode(priority, freq, pulses, timeout), _num_leds(nLeds), _toggle(false), _timer_inc(0), leds_margin(5)
+  {
+    _color = color;
+    if (_pulses != 0)
+    {
+      _pulses *= 2;
+      _pulses += 1;
+    }
+    _inc = (1. / UPDATE_RATE_HZ) * _freq;
+  }
+
+  virtual void execute()
+  {
+    std::vector<color::rgba> cols;
+    color::rgba turnedOff;
+    turnedOff.a = 0;
+
+    cols.push_back(turnedOff);
+    cols.push_back(turnedOff);
+    cols.push_back(turnedOff);
+    cols.push_back(turnedOff);
+    cols.push_back(turnedOff);
+
+    for (uint16_t i = leds_margin; i < (_num_leds / 2) - leds_margin; ++i)
+    {
+      cols.push_back(_color);
+    }
+    for (uint16_t i = (_num_leds / 2) - leds_margin; i < _num_leds; ++i)
```

Comment:
I think this is the only difference between the left and right modes, correct? 
Is there really no way to have these both derive from a common class (eg. `DirectionIndicatorMode` or something better) and have a `direction` variable set to either left/right that makes the quintessential difference between these modes?